### PR TITLE
Tags search

### PR DESCRIPTION
**Pull request by 0tpyrk**
Added the tag search functionality and tests.

Currently takes every event that has one of the tags in the search query and sorts them by the number of tags they have in common with the search query. If two events have the same number of tags, then it sorts in integer order (to work with the tests I designed) but will be replaced in the future with another level of sorting (location distance, attendee numbers, or just alphabetic).

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/Utility.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import com.google.gson.Gson;
+import java.util.List;
+import java.util.ArrayList;
+import javax.servlet.http.HttpServletRequest;
+
+
+public class Utility {
```
Comment (Yunni) [inline src/main/java/com/google/sps/servlets/Utility.java:24]: `Utils` is more commonly used.

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 2]
```diff
@@ -40,21 +53,69 @@
...
+  }
+  
+  /**
+   * Returns a count of the number of tags two lists have in common
+   *
+   * @return Integer count of number of tags in common
+   * @param tagListA List of tags to be compared
+   * @param tagListB List of tags to be compared
+   */
+  public Integer tagsInCommon(List<String> tagListA, List<String> tagListB) {
```
Comment (Yunni) [inline src/main/java/com/google/sps/servlets/SearchServlet.java:113]: Maybe the method name `intersection` is more readable?

File: src/main/java/com/google/sps/servlets/Utility.java (hunk 1 above)
Comment (0tpyrk) [reply src/main/java/com/google/sps/servlets/Utility.java:24]: Got it!

File: src/main/java/com/google/sps/servlets/SearchServlet.java (hunk 2 above)
Comment (0tpyrk) [reply src/main/java/com/google/sps/servlets/SearchServlet.java:113]: Got it!

**Pull request merged**